content/material/2020-01-17-intermediate-dplyr.Rmd

```diff
@@ -0,0 +1,333 @@
+- "Luke Johnston"
+
+date: "1/17/2020"
+description: >
+  Learn to reshape data in long or wide form, manage factors and dates using the packages dplyr, tidyr, lubridate, forcats and stringr.
+categories:
+  - Intermediate
+  - R
+  - Code-along
+tags:
+  - tidyr
+  - dplyr
+  - forcats
+  - stringr
+  - lubridate
+  - data manipulation
+  - factors
+  - dates
+---
+
+
+```{r setup, echo=FALSE}
+source(here::here("R/utils-for-content.R"))
+```
+
+## Preparation
+
+Please install the tidyverse package:
+
+```{r, eval=FALSE}
+install.packages("tidyverse")
+library(lubridate)
+```
+
+### Prerequisites
+
+In this lesson we assume a bit of familiarity with the basic functions of dplyr (especially group-by() and summarize()), which we covered in the [intro_dplyr lesson](https://au-cru.github.io/site/material/2019-11-15-intro-dplyr/) 
+
+
+### Objectives
+
+1. To learn to restructure your data between long and wide format with pivot_longer() and pivot_wider() functions from tidyr package.
+2. Learn to combine group_by() and summarize() function with pivot_longer() and pivot_wider() functions.
+3. Manage dates with lubridate functions.
+
+    
+**At the end of this session you will be able:**
+
+1. To re-structure your data in long or wide format.
+2. To reshape data by using grouping, summarizing and reshaping dataframes.
+3. To reformat dates with lubridate
+
+### Why learning to reformat and reshape data?
+
+Often we receive databases that have been created by others, and in  which information is organized in a "messy" way. Some other times, we simply need to reorder the structure of the data to be able to carry out certain data analysis or data visualization that require a specific data structure. 
+
+The tidyverse, provides some very powerful packages (e.g. dplyr, tidyr) that can allow for complete reorganization of our data in a few lines of codes. Using these packages and their functions has several advantages:
+
+- It makes our lives easier and our work faster, because we can jump between different 
+formats of our data in an agile and easy-to-track way;
+
+- It makes our work enterely reproducible.
+
+### Wide and long formats
+
+One of the most common operations that we may need to do is to change the way our data is organized from a long (few columns, many rows) to a wide (few rows, many columns) format.
+
+The functions we want to use for these operations are `pivot_longer()` and `pivot_wider()` from the tidyr package (Those of you already familiar with this package may know these functions as spread() and gather(). Although these two functions still work, they are being slowly replaced by the pivot_** functions).
+
+To show how these functions work, we are going to use the built-in dataframe `population`.
+
+#### Familiarize with the dataframe
+
+Using the function View() inspect the dataframe population, which shows the populations census in different countries across multiple years.
+
+```{r}
+View(population)
+```
+
+
+### Pivot_wider()
```

Comment:
A bit more descriptive header titles would make it easier to understand the content of the section.